packages/domain-search/src/components/domain-suggestion-cta/index.tsx

```diff
@@ -9,36 +9,47 @@ import './style.scss';
 interface DomainSuggestionCTAProps {
 	compact?: boolean;
 	uuid: string;
+	onClick?( action: 'add-to-cart' | 'continue' ): void;
 }
 
-export const DomainSuggestionCTA = ( { compact, uuid }: DomainSuggestionCTAProps ) => {
+export const DomainSuggestionCTA = ( { compact, uuid, onClick }: DomainSuggestionCTAProps ) => {
 	const { __ } = useI18n();
 	const { cart, onContinue } = useDomainSearch();
 
 	const isDomainOnCart = cart.hasItem( uuid );
 
 	if ( isDomainOnCart ) {
+		const handleContinueClick = () => {
+			onClick?.( 'continue' );
+			onContinue();
+		};
+
 		return (
 			<Button
 				variant="primary"
 				__next40pxDefaultSize
 				icon={ arrowRight }
 				className="domain-suggestion-cta domain-suggestion-cta--continue"
-				onClick={ onContinue }
+				onClick={ handleContinueClick }
 				label={ __( 'Continue' ) }
 			>
 				{ compact ? undefined : __( 'Continue' ) }
 			</Button>
 		);
 	}
 
+	const handleAddToCartClick = () => {
+		onClick?.( 'add-to-cart' );
+		cart.onAddItem( uuid );
+	};
```

Comment:
Same here. Can we memoize the function definition with useCallback here?

Reply:
The same as above 😅